src/components/List/List.jsx

```diff
@@ -1,35 +1,68 @@
-import React, { useRef, forwardRef, useMemo } from "react";
+import React, { useRef, forwardRef, useMemo, useState, useCallback } from "react";
 import PropTypes from "prop-types";
 import cx from "classnames";
 import useMergeRefs from "../../hooks/useMergeRefs";
 import { VirtualizedListItems } from "components/List/VirtualizedListItems/VirtualizedListItems";
 import "./List.scss";
+import { keyCodes } from "../../constants/KeyCodes";
 
 const List = forwardRef(
   ({ className, id, component, children, dense, ariaLabel, ariaDescribedBy, renderOnlyVisibleItems, style }, ref) => {
     const componentRef = useRef(null);
+    const [focusIndex, setFocusIndex] = useState(0);
     const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
     const Component = component;
+    const childrenRefs = useRef([]);
+    const onKeyDown = useCallback(
+      event => {
+        const isUpKey = event.keyCode === keyCodes.UP_ARROW;
+        const isDownKey = event.keyCode === keyCodes.DOWN_ARROW;
+        let overrideFocusIndex = undefined;
+        if (isUpKey || isDownKey) {
+          if (isDownKey && focusIndex + 1 < childrenRefs.current.length) {
```

Comment:
do we want to add cyclic navigation ?

Reply:
yes we do but I did not want to make it a blocker. but according to a11y requirements it should be